### ISSUE TITLE
File open fails with unicode filename

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#694]
_[Trac time 20090612 143414Z]_ A filename with non-ascii characters fails to load:

```
Traceback (most recent call last):
  File "/home/gnutella/lib/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14572, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/home/gnutella/peppy-0.10.0/peppy/mainmenu.py", line 117, in action
    self.openFiles(paths)
  File "/home/gnutella/peppy-0.10.0/peppy/mainmenu.py", line 94, in openFiles
    self.frame.open("file:%s" % path)
  File "/home/gnutella/peppy-0.10.0/peppy/frame.py", line 586, in open
    opener.open()
  File "/home/gnutella/peppy-0.10.0/peppy/fileopener.py", line 97, in open
    if self.findBufferIfAlreadyLoaded():
  File "/home/gnutella/peppy-0.10.0/peppy/fileopener.py", line 107, in findBufferIfAlreadyLoaded
    self.dprint("Looking for %s in current buffer list" % self.url)
  File "/home/gnutella/peppy-0.10.0/peppy/vfs/itools/uri/generic.py", line 554, in __str__
    path = str(self.path)
  File "/home/gnutella/peppy-0.10.0/peppy/vfs/itools/uri/generic.py", line 264, in __str__
    path += '/'.join([ str(x) for x in self ])
UnicodeEncodeError: 'ascii' codec can't encode characters in position 4-6: ordinal not in range(128)
```

but it looks like it might only be due to a dprint?

**Comment by robmcmullen** [comment]
_[Trac time 20090612 144135Z]_ I need to be using unicode(url) instead of str(url) in dprints

**Comment by robmcmullen** [comment]
_[Trac time 20090612 144435Z]_ More errors from dprint:

```
Traceback (most recent call last):
  File "/home/gnutella/lib/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14572, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/home/rob/src/peppy/peppy/mainmenu.py", line 117, in action
    self.openFiles(paths)
  File "/home/rob/src/peppy/peppy/mainmenu.py", line 94, in openFiles
    self.frame.open("file:%s" % path)
  File "/home/rob/src/peppy/peppy/frame.py", line 585, in open
    opener = FileOpener(self, *args, **kwargs)
  File "/home/rob/src/peppy/peppy/fileopener.py", line 89, in __init__
    self.dprint(self)
  File "/home/rob/src/peppy/peppy/debug.py", line 118, in dprint
    dlogfh.write("%s:%d %s.%s: %s%s" % (os.path.basename(caller[1]),caller[2],cls.__name__,caller[3],str,os.linesep))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 72-74: ordinal not in range(128)
```

**Comment by robmcmullen** [comment]
_[Trac time 20090612 170641Z]_ 03ba5fc450de106cdd420d701d659a2ab3d6be88 Fixed #694: changed debug printing that uses URLs to use unicode strings
